### ISSUE TITLE
Possible migration of a very big Excel project (xlsx and xlam) to jExcel

### ISSUE THREAD

**Issue by cavo789** [jspreadsheet/ce#1301]
Hi jExcel,

Since 2013, I develop and maintain a client-server application (Excel for the interface and SQL Server for the backend). My code is developed in VBA for the Excel part and consists of several tens of thousands of programming lines.

We are in 2021 and I wish to investigate the possibility of rewriting the Excel part in a web version; php (probably Laravel) and VueJs. 

I quickly went through your examples and they speak for themselves to get a quick idea of the possibilities.

However, as my application is extremely complex and multiple (about ten Excel interfaces); I'm trying to get a quick idea of the limitations of jExcel compared to Excel itself (heavy client version (.exe application); not Excel online).

Do you have any documents listing the limitations of jExcel, i.e. the existing functionalities in Excel that are currently missing in jExcel?

* https://jexcel.net/v7/
* https://github.com/jspreadsheet/jexcel

To be precise: at the moment, I’m just asking myself if jExcel could be a way to rewrite the application in a more modern version (web) than an .xslm file.

The web application that would be developed will have about a hundred users; several profiles will exist (administrator, power users, ...).

Thanks

**Comment by GBonnaire** [comment]
Hi,

For share my experience with JExcel Pro, in start i use XLSM, and next i use Google Sheet, and i ask same question of you. Currently, i use in different case in my app. (Planning, pivot, interaction with user, print options, ...) writed in JS/HTML/PHP/MySQL

Careful : currently, Jexcel CE (free) different JExcel Pro (paid), but with pro more options, features, and can use more rows and quick dev support.

Main advantages of JExcel : 
- LazyLoading, when you have lot of row. My sheets it's with min 4 000 cells to 30 000 rows. i use JS Pure for JExcel and some plugin JQuery for custom editor.
- Plugins : make your feature with plugin, i use this feature all time with 5 to 10 plugins per sheet. i create lot of plugins for my app, some example of plugins it's available in ( https://jexcel.net/plugins/ and https://demo.gbonnaire.fr and https://github.com/Guillaume-Bo/jexcel-plugins-and-editors)
- Cloud/persistance options : You can easier send informations in server for persistence, i make my plugin cloud for make with my app, but with Jexcel cloud, you can use that it's more simple. And currently in my app, i have more 500 users not technical (not in same sheets, but sometime with 4/5 users edit same file in same time with different right, and it's work very fine).

I think @paulhodel can answer better to your questions

have good day

**Comment by cavo789** [comment]
Bonjour et merci Guillaume

For sure, I'll start with jExcel CE just to have an idea of what is possible then will upgrade very quickly to jExcel PRO as soon as I can get a better idea of what is possible within my project's constraints.

Thanks to have take time to list a few advantages and share your experience.

My current application (Excel VBA thus) is used by probably 10 users concurrently. They can fill in text or figures in cells; typing the value or using lists f.i.. Once they've filled in the "sheet", today, they've a Submit button in a custom ribbon. The code behind will check any modified rows and then any modified cells on these rows. For each row, a SQL statement will be prepared to send the changes to my SQL Server db. I don't need any cloud here; just be able to access my backend DB (easy with PHP).

My application has "grosso-modo" 60,000 lines of VBA code in two addins (xlam) and in the interface itself (xlsm) so, at this time, I'm worry about : is jExcel already mature enough to be able to replace my VBA / Excel client application and be able to support my features.

I am aware that I am not very precise, but what are the current limitations of jExcel? What is it not yet able to do in terms of interface? What do I need to be aware of if I launch a proof-of-concept? I wouldn't want to spend three months coding with jExcel only to realise that, no, this or that functionality is missing when my project relies on it. 

Thanks by advance.

(I'll take a look to your demo site and plugins, many thanks)

**Comment by GBonnaire** [comment]
Bonjour, pas de quoi (c'est plus simple en français 😋 mais bon faut partager en anglais 😊  )

For precise : You can develop for free with jExcel v7 pro if you develop in localhost. (and all my plugin it's available in demo in localhost)

But, maybe wait some weeks, new version of jExcel should be publish and more options who interest you.

For limit, currently in my case, i'm not touch limit, it's work fine with the big data. Maybe @paulhodel can informed you on limit of jExcel pro. I know in version of CE it's more limited (more slowly with big data).
For interface, it use ajax post to send data in post to server or you can sync jExcel with input and manage that as an traditional HTML form.
For POC, you should change his way of thinking with traditional Excel, it's different way. You can use Jexcel like 1 sheet offline and each change, modify an input value who contains JSON of jExcel (Behavior more closer of Excel), or to have interactive sheet (send to server each change). The conception it's totaly different, but possible with jExcel.

With Jexcel Pro, you can have more assistance for help you when you have blocked, with Paul or Me or Other.

After, i can't say more, i don't know your app, but i know more possibility with plugins. You have see my plugins and look different possibilities.

For example, in my case : i use Meta data in each cells, each cells contains the informations when it's change, with persistance options (and 1 plugin), it send on server with this meta data, and on server i change only this data.
If you have more user connected in same time on sheet, i think to use private cloud or equiv (in my case, i use get interval for get last change and set this change on sheet with 5sec interval).

**Comment by hodeware** [comment]
Hi,

If you are going to create a complex project my suggestion is to go straight to the Pro version. You can test it fully for as long as you need with all features enabled for free. You create a plan when you are ready to go to production.

There are so many reasons for that, starting with the smart DOM management engine, cell-level properties support, multiple worksheets and so many other extensions and features in sorting, filtering, searching, pagination, lazyLoading, and persistence.

Basically, you will have better performance and will create more writing less. With the Plugins and Rich editors, you are able to go beyond the native features, creating your own custom behavior.

With the PRO support, it will save hours in researching because we can point you in the right direction via skype or zoom. If you would like to save time you should definitely start on the v7 version.

Regards,

**Comment by cavo789** [comment]
Thank you guys.

I'm actually still busy on two ICT projects so I'll not start in the next months but really excited to. If I receive the GO of my management, I'll then start coding a proof of concept. Probably in six months.

Many thanks for the info and for the tip to start with the pro (fully working on localhost, that's great !).
